node/pod.go

```diff
@@ -70,7 +71,7 @@ func (pc *PodController) createOrUpdatePod(ctx context.Context, pod *corev1.Pod)
 	if podFromProvider, _ := pc.provider.GetPod(ctx, pod.Namespace, pod.Name); podFromProvider != nil {
 		if !podsEqual(podFromProvider, podForProvider) {
 			log.G(ctx).Debugf("Pod %s exists, updating pod in provider", podFromProvider.Name)
-			if origErr := pc.provider.UpdatePod(ctx, podForProvider); origErr != nil {
+			if origErr := pc.provider.UpdatePod(ctx, podForProvider); origErr != nil && !errdefs.IsUnsupported(origErr) {
```

Comment:
Can we make sure not to log that the pod was updated in the unsupported case?

We may also want to have the queue handler deal with unsupported by just not re-queuing instead of returning no error. WDYT?

Reply:
I'm worried if we do it in the queue handler for all queues, then it'll be unable to differentiate isunsupported from things which are allowed to be unsupported versus those which are not.

Maybe we should do something like K8s, and have the ability to signal how retryable errors are? https://github.com/kubernetes/apimachinery/blob/master/pkg/api/errors/errors.go#L619

Reply:
When should an `Unsuported` error be ok to re-queue?

Reply:
Never, but no provider should ever have something like CreatePod unsupported.

Reply:
No working provider, but it will be obvious that things aren't working by looking at the logs (And the fact that nothing works :) ).

Reply:
I still think we should return the error here.
The provider didn't perform the action and callers need to know that information.